rplidar.go

```diff
@@ -35,16 +32,47 @@ const (
 // Model is the model of the rplidar
 var Model = resource.NewModel("viam", "lidar", "rplidar")
 
+// Rplidar controls an Rplidar device.
+type Rplidar struct {
+	resource.Named
+	resource.AlwaysRebuild
+	mu                      sync.Mutex
+	device                  rplidarDevice
+	nodes                   gen.Rplidar_response_measurement_node_hq_t
+	nodeSize                int
+	started                 bool
+	scannedOnce             bool
+	defaultNumScans         int
+	warmupNumDiscardedScans int
+
+	logger golog.Logger
+}
+
+// Config describes how to configure the RPlidar component.
+type Config struct {
+	DevicePath string `json:"device_path"`
+}
+
+// Validate checks that the config attributes are valid for an RPlidar.
+func (conf *Config) Validate(path string) ([]string, error) {
+	return nil, nil
+}
+
 func init() {
-	registry.RegisterComponent(camera.Subtype, Model, registry.Component{Constructor: newRplidar})
+	resource.RegisterComponent(camera.API, Model, resource.Registration[camera.Camera, *Config]{Constructor: newRplidar})
 }
 
-func newRplidar(ctx context.Context, _ registry.Dependencies, c config.Component, logger golog.Logger) (interface{}, error) {
-	devicePath := c.Attributes.String("device_path")
+func newRplidar(ctx context.Context, _ resource.Dependencies, c resource.Config, logger golog.Logger) (camera.Camera, error) {
+	svcConf, err := resource.NativeConfig[*Config](c)
+	if err != nil {
+		return nil, err
+	}
+
+	devicePath := svcConf.DevicePath
 	if devicePath == "" {
 		var err error
 		if devicePath, err = searchForDevicePath(logger); err != nil {
-			return config.Component{}, errors.Wrap(err, "need to specify a devicePath (ex. /dev/ttyUSB0)")
+			return nil, errors.Wrap(err, "need to specify a devicePath (ex. /dev/ttyUSB0)")
```

Comment:
This was returning the wrong type, which wasn't caught as before `newRplidar` returned an empty interface type, now b/c it returns a `camera.Camera` this bug was caught by the compiler.